app/models/course/condition/assessment.rb

```diff
@@ -0,0 +1,34 @@
+class Course::Condition::Assessment < ActiveRecord::Base
+  include ActiveSupport::NumberHelper
+
+  acts_as_condition
+  belongs_to :assessment, class_name: Course::Assessment.name, inverse_of: false
+
+  default_scope { includes(:assessment) }
+
+  def title
+    if minimum_grade_percentage
+      "#{assessment.title} (at least "\
+      "#{number_to_percentage(minimum_grade_percentage,  precision: 2,
+                              strip_insignificant_zeros: true)})"
+    else
+      assessment.title
+    end
+  end
+
+  def satisfied_by?(course_user)
```

Comment:
@minhtule I mean the entire function, not the line. There are many returns within, they can probably be split into functions that check for one thing each then boolean AND all of them.